### PR TITLE
Restructure docs

### PR DESCRIPTION
**Pull request by yoshi-1224**
Fixes issue #198 

## Proposed Changes
  - move most of the stuff in README to docs/
  - use mkdocs (material theme?)
 
Mkdocs website can be tried out by:
```
# from project root
pip install mkdocs mkdocs-material pymdown-extensions
mkdocs build
mkdocs serve
```
This is definitely going to be an iterative process, so please let me know how I can improve the structure/writing!

### CODE REVIEW

File: docs/secrets.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+This will substitute the referenced secrets with the actual decrypted secrets stored at the referenced paths and display the file content.
+
+## Secret sub-variables
+
+As illustrated above, one file corresponds to one secret. It is now possible for users who would like to reduce the decryption overhead to manually create a yaml file that contains multiple secrets, each of which can be referenced by its object key. For example, consider the secret file `secrets/mysql_secrets`:
+
+```yaml
+mysql_passwords:
+	secret_foo: hello_world
```
Comment (uberspot) [inline docs/secrets.md]: These are marked as tabs in github. This might be a good chance to change these to just spaces as we have everywhere else. :)

**Comment by yoshi-1224** [comment]
@uberspot @ramaro 
Is there anything else on the docs that I should add or modify about? Should I update the part on secrets to reflect the upcoming change (i.e. change into refs)?

**Comment by uberspot** [comment]
No that's ok, that can go in the same PR as the refs work. :) I'd consider this good to go.

**Comment by ramaro** [comment]
@yoshi-1224 this is great! Good to go.

**Comment by uberspot** [comment]
@yoshi-1224 it only needs a rebase to fix conflicts. :)

**Comment by yoshi-1224** [comment]
Hi @uberspot @ramaro 
Thanks for the reviews! Just fixed the issue. 
I also put a link to https://kapitan.dev under the documentation heading on README so that people visit that site rather than our docs/ folder. I don't think I have access to https://kapitan.dev so I guess this PR is done on my side? I believe the deployment can be performed as described in my first comment. Let me know if it doesn't

**Comment by uberspot** [comment]
Doc deployed here https://kapitan.dev :) 
Nice work @yoshi-1224 ! 
If you spot any broken links etc feel free to send a PR. 
After any change to docs/ we also need to do `mkdocs gh-deploy -b my_gh_branch_name` to push the changes to my_gh_branch_name and then do a PR to merge to gh-pages. 
I've added a branch rule to enforce that as well.

**Pull request merged**